In [ ]:
!pip install googletrans==3.1.0a0
!pip install fairseq
!pip install sentencepiece

  Preparing metadata (setup.py) ... done
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 55.1/55.1 kB 3.0 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 11.2 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 133.4/133.4 kB 11.3 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.8/58.8 kB 9.4 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 42.6/42.6 kB 6.3 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 53.6/53.6 kB 8.1 MB/s eta 0:00:00
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 65.0/65.0 kB 10.0 MB/s eta 0:00:00
  Created wheel for googletrans: filename=googletrans-3.1.0a0-py3-none-any.whl size=16352 sha256=7a222b701d5e7450748f749817d77bb3d50c86d81cb8f271d31edb13cb363315
  Stored in directory: /root/.cache/pip/wheels/50/5d/3c/8477d0af4ca2b8b1308812c09f1930863caeebc762fe265a95
Successfully built googletrans
  Attempting uninstall: chardet
    Found existing installation: chardet 5.2.0
    Uninstalling ch

In [ ]:
from google.colab import drive
# Mount from drive
drive.mount('/content/drive')

Mounted at /content/drive


In [ ]:
!git clone https://github.com/google/sentencepiece.git
!mkdir sentencepiece/build
!cd sentencepiece/build && cmake ..
!cd sentencepiece/build && make -j $(nproc)

Cloning into 'sentencepiece'...
remote: Enumerating objects: 4823, done.
remote: Counting objects: 100% (1886/1886), done.
remote: Compressing objects: 100% (212/212), done.
remote: Total 4823 (delta 1701), reused 1674 (delta 1674), pack-reused 2937
Receiving objects: 100% (4823/4823), 26.68 MiB | 16.45 MiB/s, done.
Resolving deltas: 100% (3338/3338), done.
CMake Deprecation Warning at CMakeLists.txt:15 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- VERSION: 0.2.00
-- The C compiler identification is GNU 11.4.0
-- The CXX compiler identification is GNU 11.4.0
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX c

# Dataset

The test dataset using is IWSLT'15 English-Vietnamese which is available at https://nlp.stanford.edu/projects/nmt/

In [ ]:
# Download test data
!wget https://nlp.stanford.edu/projects/nmt/data/iwslt15.en-vi/tst2012.en
!wget https://nlp.stanford.edu/projects/nmt/data/iwslt15.en-vi/tst2012.vi
!wget https://nlp.stanford.edu/projects/nmt/data/iwslt15.en-vi/tst2013.en
!wget https://nlp.stanford.edu/projects/nmt/data/iwslt15.en-vi/tst2013.vi

--2023-09-07 13:02:43--  https://nlp.stanford.edu/projects/nmt/data/iwslt15.en-vi/tst2012.en
Resolving nlp.stanford.edu (nlp.stanford.edu)... 171.64.67.140
Connecting to nlp.stanford.edu (nlp.stanford.edu)|171.64.67.140|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 140250 (137K) [text/plain]
Saving to: ‘tst2012.en’

tst2012.en          100%[===================>] 136.96K  --.-KB/s    in 0.1s    

2023-09-07 13:02:43 (1.41 MB/s) - ‘tst2012.en’ saved [140250/140250]

--2023-09-07 13:02:43--  https://nlp.stanford.edu/projects/nmt/data/iwslt15.en-vi/tst2012.vi
Resolving nlp.stanford.edu (nlp.stanford.edu)... 171.64.67.140
Connecting to nlp.stanford.edu (nlp.stanford.edu)|171.64.67.140|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 188396 (184K) [text/plain]
Saving to: ‘tst2012.vi’

tst2012.vi          100%[===================>] 183.98K  --.-KB/s    in 0.1s    

2023-09-07 13:02:44 (1.88 MB/s) - ‘tst2012.vi’ saved [188396/188396]

--2023

# Evaluate test set with googleTrans

In [ ]:
from googletrans import Translator
from tqdm import tqdm
import fileinput

#define a translate function
def translate_GoogleTrans(vi_lst, ver):
  translator = Translator()
  with open("googleTrans_" + str(ver) +".en", "w") as file_trans:
    for vi in tqdm(vi_lst):
      result = translator.translate(vi, src='vi', dest='en')
      print(result.text, file=file_trans)

Translate vi to en on testset tst2012.vi and tst2013.vi using GoogleTrans

In [ ]:
#Read test set 2012 and translate
vi_2012 = [vi[:-1] for vi in open('/content/tst2012.vi','r')]
translate_GoogleTrans(vi_2012, 2012)

100%|██████████| 1553/1553 [00:49<00:00, 31.60it/s]


In [ ]:
# Display 10 translated sentence from vi - en on tst2012
!head -n 10 googleTrans_2012.en

How can I present in 10 minutes about the bond that binds women through three generations, about how those amazingly strong bonds have gripped the life of a four-year-old girl? curled up with her little sister , with her mother and grandmother during the five days and nights on a small boat floating in the East Sea more than 30 years ago , the bonds that held her life and never now leave -- that girl now lives in San Francisco and is talking to you today ?
This story is not over.
It is a jigsaw puzzle that is still being stacked .
Let me tell you about some of the pieces.
Imagine the first piece: a man who burned his career for the rest of his life.
He was a poet and playwright, a man whose whole life was precarious with the only hope that his country would be independent and free.
Imagine him, a communist entering Saigon, facing the fact that his whole life was wasted.
Words, which have been his companion for many years, are now mocking him.
He retreated into silence.
He died, knocked

In [ ]:
#Read test set 2013 and translate
vi_2013 = [vi[:-1] for vi in open('/content/tst2013.vi','r')]
translate_GoogleTrans(vi_2013, 2013)

100%|██████████| 1268/1268 [00:44<00:00, 28.50it/s]


In [ ]:
# Display 10 translated sentence from vi - en on tst2013
!head -n 10 googleTrans_2013.en

When I was little, I thought North Korea was the best country in the world and I used to sing "We have nothing to be jealous of."
I was very proud of my country.
At school, we spend a lot of time learning about the life of president Kim II-Sung, but we don't learn much about the outside world, except that the United States, Korea and Japan are our enemies. I .
Although I used to wonder what the outside world was like, I still thought that I would live my whole life in North Korea, until everything suddenly changed.
When I was 7 years old, I witnessed people being shot publicly for the first time in my life, but I still think my life here is completely normal.
My family is not poor, and I myself have never suffered from hunger.
But one day in 1995 , my mother brought home a letter from a sister in her workplace .
In it is written: When you read these lines, my whole family of 5 is no longer in this world, because my whole family has had nothing to eat for two weeks.
We were all lying on

Calculate bleu score on 2 translated set

In [ ]:
# Calculate on tst2012
!sacrebleu -tok 'none' -s 'none' tst2012.en < googleTrans_2012.en

sacreBLEU: That's 100 lines that end in a tokenized period ('.')
sacreBLEU: It looks like you forgot to detokenize your test data, which may hurt your score.
sacreBLEU: If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
{
 "name": "BLEU",
 "score": 21.7,
 "signature": "nrefs:1|case:mixed|eff:no|tok:none|smooth:none|version:2.3.1",
 "verbose_score": "56.6/31.8/19.5/12.3 (BP = 0.845 ratio = 0.856 hyp_len = 23956 ref_len = 27983)",
 "nrefs": "1",
 "case": "mixed",
 "eff": "no",
 "tok": "none",
 "smooth": "none",
 "version": "2.3.1"
}


In [ ]:
# Calculate on tst2013
!sacrebleu -tok 'none' -s 'none' tst2013.en < googleTrans_2013.en

sacreBLEU: That's 100 lines that end in a tokenized period ('.')
sacreBLEU: It looks like you forgot to detokenize your test data, which may hurt your score.
sacreBLEU: If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
{
 "name": "BLEU",
 "score": 25.8,
 "signature": "nrefs:1|case:mixed|eff:no|tok:none|smooth:none|version:2.3.1",
 "verbose_score": "60.9/36.4/23.3/15.6 (BP = 0.860 ratio = 0.869 hyp_len = 23217 ref_len = 26728)",
 "nrefs": "1",
 "case": "mixed",
 "eff": "no",
 "tok": "none",
 "smooth": "none",
 "version": "2.3.1"
}


# Evaluate test set with our model

In [ ]:
# SPM encoding
%env SPM=/content/sentencepiece/build/src
!mkdir tokenized
!$SPM/spm_encode --model="drive/MyDrive/sentencepiece.bpe.model" --output_format=piece < "tst2012.vi" > tokenized/tst2012.spm.vi
!$SPM/spm_encode --model="drive/MyDrive/sentencepiece.bpe.model" --output_format=piece < "tst2013.vi" > tokenized/tst2013.spm.vi

env: SPM=/content/sentencepiece/build/src


In [ ]:
#Preprocessing for tst2012
!fairseq-preprocess \
    --only-source \
    --source-lang vi --target-lang en \
    --joined-dictionary \
    --srcdict drive/MyDrive/checkpointbest/dict.vi.txt \
    --testpref tokenized/tst2012.spm \
    --destdir data-bin-2012 \
    --workers 70;

2023-09-07 13:30:26.905395: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-07 13:30:27.777133: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2023-09-07 13:30:29 | INFO | fairseq.tasks.text_to_speech | Please install tensorboardX: pip install tensorboardX
2023-09-07 13:30:29 | INFO | fairseq_cli.preprocess | Namespace(no_progress_bar=False, log_interval=100, log_format=None, log_file=None, aim_repo=None, aim_run_hash=None, tensorboard_logdir=None, wandb_project=None, azureml_logging=False, seed=1, cpu=False, tpu=False, bf16=False, memory_efficient_bf16=False, fp16=False, memory_efficient_fp16=False, fp16_no_flatten_grads=False, fp16_init_scale=128, fp16_scale_window=None, fp16_scale_tolerance=0.0,

In [ ]:
#Preprocessing for tst2013
!fairseq-preprocess \
    --only-source \
    --source-lang vi --target-lang en \
    --joined-dictionary \
    --srcdict drive/MyDrive/checkpointbest/dict.vi.txt \
    --testpref tokenized/tst2013.spm \
    --destdir data-bin-2013 \
    --workers 70;

2023-09-07 13:31:04.003386: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-07 13:31:05.329614: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2023-09-07 13:31:07 | INFO | fairseq.tasks.text_to_speech | Please install tensorboardX: pip install tensorboardX
2023-09-07 13:31:07 | INFO | fairseq_cli.preprocess | Namespace(no_progress_bar=False, log_interval=100, log_format=None, log_file=None, aim_repo=None, aim_run_hash=None, tensorboard_logdir=None, wandb_project=None, azureml_logging=False, seed=1, cpu=False, tpu=False, bf16=False, memory_efficient_bf16=False, fp16=False, memory_efficient_fp16=False, fp16_no_flatten_grads=False, fp16_init_scale=128, fp16_scale_window=None, fp16_scale_tolerance=0.0,

In [ ]:
# Copy en dict to 2012 and 2013 data-bin-dir
!cp drive/MyDrive/checkpointbest/dict.en.txt data-bin-2012/
!cp drive/MyDrive/checkpointbest/dict.en.txt data-bin-2013/

In [ ]:
# Generate on 2012 test set
!fairseq-generate \
    data-bin-2012 \
  --path drive/MyDrive/checkpointbest/checkpoint_best.pt \
  --task translation_from_pretrained_bart \
  --source-lang vi --target-lang en \
  --bpe 'sentencepiece' --sentencepiece-model drive/MyDrive/sentencepiece.bpe.model \
  --sacrebleu --remove-bpe 'sentencepiece' \
  --beam 5 --nbest 1 \
  --batch-size 32 --langs ar_AR,cs_CZ,de_DE,en_XX,es_XX,et_EE,fi_FI,fr_XX,gu_IN,hi_IN,it_IT,ja_XX,kk_KZ,ko_KR,lt_LT,lv_LV,my_MM,ne_NP,nl_XX,ro_RO,ru_RU,si_LK,tr_TR,vi_VN,zh_CN > result_2012.out


2023-09-07 13:38:09.249104: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-07 13:38:10.187828: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2023-09-07 13:38:12 | INFO | fairseq.tasks.text_to_speech | Please install tensorboardX: pip install tensorboardX
2023-09-07 13:38:16 | INFO | fairseq_cli.generate | {'_name': None, 'common': {'_name': None, 'no_progress_bar': False, 'log_interval': 100, 'log_format': None, 'log_file': None, 'aim_repo': None, 'aim_run_hash': None, 'tensorboard_logdir': None, 'wandb_project': None, 'azureml_logging': False, 'seed': 1, 'cpu': False, 'tpu': False, 'bf16': False, 'memory_efficient_bf16': False, 'fp16': False, 'memory_efficient_fp16': False, 'fp16_no_flatten_grad

In [ ]:
# Generate on 2013 test set
!fairseq-generate \
    data-bin-2013 \
  --path drive/MyDrive/checkpointbest/checkpoint_best.pt \
  --task translation_from_pretrained_bart \
  --source-lang vi --target-lang en \
  --bpe 'sentencepiece' --sentencepiece-model drive/MyDrive/sentencepiece.bpe.model \
  --sacrebleu --remove-bpe 'sentencepiece' \
  --beam 5 --nbest 1 \
  --batch-size 32 --langs ar_AR,cs_CZ,de_DE,en_XX,es_XX,et_EE,fi_FI,fr_XX,gu_IN,hi_IN,it_IT,ja_XX,kk_KZ,ko_KR,lt_LT,lv_LV,my_MM,ne_NP,nl_XX,ro_RO,ru_RU,si_LK,tr_TR,vi_VN,zh_CN > result_2013.out

2023-09-07 13:41:30.008175: I tensorflow/core/platform/cpu_feature_guard.cc:182] This TensorFlow binary is optimized to use available CPU instructions in performance-critical operations.
To enable the following instructions: AVX2 FMA, in other operations, rebuild TensorFlow with the appropriate compiler flags.
2023-09-07 13:41:31.562161: W tensorflow/compiler/tf2tensorrt/utils/py_utils.cc:38] TF-TRT Warning: Could not find TensorRT
2023-09-07 13:41:34 | INFO | fairseq.tasks.text_to_speech | Please install tensorboardX: pip install tensorboardX
2023-09-07 13:41:36 | INFO | fairseq_cli.generate | {'_name': None, 'common': {'_name': None, 'no_progress_bar': False, 'log_interval': 100, 'log_format': None, 'log_file': None, 'aim_repo': None, 'aim_run_hash': None, 'tensorboard_logdir': None, 'wandb_project': None, 'azureml_logging': False, 'seed': 1, 'cpu': False, 'tpu': False, 'bf16': False, 'memory_efficient_bf16': False, 'fp16': False, 'memory_efficient_fp16': False, 'fp16_no_flatten_grad

Calculate bleu score on 2 translated set

In [ ]:
# Calculate on tst2012
!cat result_2012.out | grep -P "^H" | sort -V | cut -f 3- | sed 's/\[en\]//g' > vi_en_2012.hyp
!sacrebleu -tok 'none' -s 'none' tst2012.en < vi_en_2012.hyp

sacreBLEU: That's 100 lines that end in a tokenized period ('.')
sacreBLEU: It looks like you forgot to detokenize your test data, which may hurt your score.
sacreBLEU: If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
{
 "name": "BLEU",
 "score": 27.2,
 "signature": "nrefs:1|case:mixed|eff:no|tok:none|smooth:none|version:2.3.1",
 "verbose_score": "60.2/33.9/21.2/13.7 (BP = 0.981 ratio = 0.981 hyp_len = 27458 ref_len = 27983)",
 "nrefs": "1",
 "case": "mixed",
 "eff": "no",
 "tok": "none",
 "smooth": "none",
 "version": "2.3.1"
}


In [ ]:
# Calculate on tst2013
!cat result_2013.out | grep -P "^H" | sort -V | cut -f 3- | sed 's/\[en\]//g' > vi_en_2013.hyp
!sacrebleu -tok 'none' -s 'none' tst2013.en < vi_en_2013.hyp

sacreBLEU: That's 100 lines that end in a tokenized period ('.')
sacreBLEU: It looks like you forgot to detokenize your test data, which may hurt your score.
sacreBLEU: If you insist your data is detokenized, or don't care, you can suppress this message with the `force` parameter.
{
 "name": "BLEU",
 "score": 33.3,
 "signature": "nrefs:1|case:mixed|eff:no|tok:none|smooth:none|version:2.3.1",
 "verbose_score": "65.6/40.2/26.7/18.2 (BP = 0.990 ratio = 0.990 hyp_len = 26463 ref_len = 26728)",
 "nrefs": "1",
 "case": "mixed",
 "eff": "no",
 "tok": "none",
 "smooth": "none",
 "version": "2.3.1"
}


# Make a comparision translated sentences between mBart, googleTrans, and the reference from test set

Finally, we got:

- **21.7** BLEU socre by GoogleTrans and **27.2** by mBart on test set 2012

- **25.8** BLEU score by GoogleTrans and **33.3** by mBart on test set 2013


In [ ]:
import pandas as pd
# Compare on 2012 test set
df_2012 = pd.DataFrame.from_dict(
    {
        'mBart': [line[:-1] for line in open('/content/vi_en_2012.hyp','r')],
        'googleTrans': [line[:-1] for line in open('/content/googleTrans_2012.en','r')],
        'reference' : [line[:-1] for line in open('/content/tst2012.en','r')]
    }
)
df_2012.head(50)

mBart  \
0   How can I present in 10 minutes the link wire ...   
1                       This story isn &apos;t over .   
2   It &apos;s a puzzle that &apos;s still being s...   
3     Let me tell you guys about a couple of pieces .   
4   Imagine the first piece : a man burning career...   
5   He was a poet , a playwright , a person whose ...   
6   Imagine him , a communist coming into Saigon ,...   
7   The word , over so many years of being a compa...   
8                         He retreated into silence .   
9                   He died , overturned by history .   
10                            He was my grandfather .   
11                       I never met him in my life .   
12  But our lives are much more than we store in o...   
13  My grandmother never allowed me to forget his ...   
14  My mission was not to let her life pass in hop...   
15  The next piece of the picture is a boat in the...   
16  My mother , Mai , was only 18 when her three l...   
17  With her mother , her life hinged on the only ...   
18  Mom never accepted that she wouldn &apos;t suc...   
19  So after four years , a field of sky even clea...   

                                          googleTrans  \
0   How can I present in 10 minutes about the bond...   
1                             This story is not over.   
2   It is a jigsaw puzzle that is still being stac...   
3           Let me tell you about some of the pieces.   
4   Imagine the first piece: a man who burned his ...   
5   He was a poet and playwright, a man whose whol...   
6   Imagine him, a communist entering Saigon, faci...   
7   Words, which have been his companion for many ...   
8                          He retreated into silence.   
9                   He died, knocked down by history.   
10                              He is my grandfather.   
11                 I have never met him in real life.   
12  But our life is much more than what we keep in...   
13  My grandmother never allowed me to forget his ...   
14  My job was not to let that life pass in vain, ...   
15  The next piece of the picture is a boat in the...   
16  My mother , Mai , was 18 years old when her fa...   
17  For her, life boils down to a single mission: ...   
18        I can never accept that I will not succeed.   
19  So, after four years, in a paradise more exten...   

                                            reference  
0   How can I speak in 10 minutes about the bonds ...  
1                      This is not a finished story .  
2    It is a jigsaw puzzle still being put together .  
3          Let me tell you about some of the pieces .  
4   Imagine the first piece : a man burning his li...  
5   He is a poet , a playwright , a man whose whol...  
6   Imagine him as the communists enter Saigon , c...  
7   Words , for so long his friends , now mocked h...  
8                         He retreated into silence .  
9                         He died broken by history .  
10                             He is my grandfather .  
11                    I never knew him in real life .  
12    But our lives are much more than our memories .  
13      My grandmother never let me forget his life .  
14  My duty was not to allow it to have been in va...  
15  The next piece of the jigsaw is of a boat in t...  
16  My mother , Mai , was 18 when her father died ...  
17  For her , life had distilled itself into one t...  
18  It was inconceivable to her that she would not...  
19  So after a four-year saga that defies fiction ...

No charts were generated by quickchart


In [ ]:
# Compare on 2013 test set
df_2013 = pd.DataFrame.from_dict(
    {
        'mBart': [line[:-1] for line in open('/content/vi_en_2013.hyp','r')],
        'googleTrans': [line[:-1] for line in open('/content/googleTrans_2013.en','r')],
        'reference' : [line[:-1] for line in open('/content/tst2013.en','r')]
    }
)
df_2013.head(50)

mBart  \
0   When I was a kid , I thought North-North Korea...   
1                      I was so proud of my country .   
2   At school , we spent a lot of time learning ab...   
3   Although I used to wonder not how the world wa...   
4   When I got to 7 , I witnessed people openly fi...   
5   My family was not poor , and I myself had neve...   
6   But on one day in 1995 , my mother brought hom...   
7   It reads : When she read these lines , then my...   
8   All lying on the floor , and our bodies gettin...   
9                                     I was shocked .   
10  Because it was the first time I knew that my f...   
11  Not long after that , as I passed a station , ...   
12  On the platform of the station was the dead bo...   
13  But there &apos;s no one to help them , becaus...   
14  In the mid- &apos; 90s , North Korea experienc...   
15  It caused over a million North Koreans to die ...   
16  The power outages were happening more and more...   
17  I was wondering all the time why they had elec...   
18  This is a picture from a satellite of North Ko...   
19  This is the Pearl River it &apos;s the natural...   
20  It &apos;s possible to see that the river bed ...   
21                          But so many people died .   
22     And I saw their bodies floating in the river .   
23  I can &apos;t say specifically how I fled Nort...   
24  At that time , I just thought I was going to h...   
25  chứ never could I imagine that I would have to...   
26  In China , it &apos;s very difficult for a gir...   
27  I hadn &apos;t imagined what it would be like ...   
28  I have always lived in a constant fear that my...   
29  One day , that nightmare came true , I was arr...   
30  Somebody informed them that I was North Korean...   
31  I was terrified , and it felt like my heart wa...   
32  For if as they saw there was anything unnatura...   
33  I thought my life coming here was ending , but...   
34  After he finished asking , one of the two poli...   
35               It &apos;s not North Korean . &quot;   
36          And they released me . It was a miracle .   
37  Some North Koreans in China went to foreign em...   
38                      these girls were very lucky .   
39  because despite being arrested , learning was ...   
40         But these North Koreans are not so lucky .   
41  Every year , there are countless North Koreans...   
42  While I was very lucky that I was released , s...   
43  The fact that North Koreans had to hide their ...   
44  Even if they &apos;ve learned Chinese and foun...   
45  That &apos;s why after 10 years of hiding my r...   
46  It &apos;s much more difficult to settle down ...   
47  Since in South Korea , English is incredibly i...   
48  I also realized a huge gap between the South a...   
49  We were all North Koreans , but became very di...   

                                          googleTrans  \
0   When I was little, I thought North Korea was t...   
1                     I was very proud of my country.   
2   At school, we spend a lot of time learning abo...   
3   Although I used to wonder what the outside wor...   
4   When I was 7 years old, I witnessed people bei...   
5   My family is not poor, and I myself have never...   
6   But one day in 1995 , my mother brought home a...   
7   In it is written: When you read these lines, m...   
8   We were all lying on the floor, and our bodies...   
9                                      I was socked .   
10  For that was the first time I knew that my com...   
11  Not long after, as I passed a train station, I...   
12  On the station floor lay the dead body of a wo...   
13  But there is no one to help them, because ever...   
14  In the mid-1990s, North Korea experienced a se...   
15  It caused more than a million Koreans to starv...   
16  Power outages were becoming more and more freq...   
17  I always wonder why they have electricity and ...   
18  This is a satellite image of North Korea at ni...   
